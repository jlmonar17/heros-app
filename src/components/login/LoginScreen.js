import { useContext } from "react";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";

export const LoginScreen = ({ history }) => {
    const { dispatch } = useContext(AuthContext);

    const handleLogin = () => {
        // Redirects, but it doesn't lose routes history
        // history.push("/");

        dispatch({
            type: types.login,
            payload: {
                name: "Jos√©",
            },
        });

        // Redirects and replace in history that /login was not visited
        history.replace("/");
    };

    return (
        <>
            <h1>LoginScreen</h1>
            <hr />

            <button className="btn btn-primary" onClick={handleLogin}>
                Login
            </button>
        </>
    );
};
