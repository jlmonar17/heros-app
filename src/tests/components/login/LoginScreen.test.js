import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";
import { LoginScreen } from "../../../components/login/LoginScreen";
describe("Tests for <LoginScreen />", function () {
    const contextValue = {
        user: { logged: false },
        dispatch: jest.fn(),
    };

    const history = { replace: jest.fn() };

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={history} />
        </AuthContext.Provider>
    );

    test("Should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Should call dispatch and navigation (history.replace)", () => {
        const handleClick = wrapper.find("button").prop("onClick");

        handleClick();

        expect(contextValue.dispatch).toHaveBeenCalled();
        expect(history.replace).toHaveBeenCalled();
        expect(history.replace).toHaveBeenCalledWith("/");

        localStorage.setItem("lastPath", "/dc");
        handleClick();
        expect(history.replace).toHaveBeenCalledWith("/dc");
    });
});
