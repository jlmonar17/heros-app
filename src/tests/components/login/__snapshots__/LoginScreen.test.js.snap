// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <LoginScreen /> Should render correctly 1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
